File: sqlLED.py
import pymysql
import RPi.GPIO as GPIO
import time

# Set pin numbering mode
GPIO.setmode(GPIO.BCM)
GPIO.setwarnings(False)

# Set pin 21 to output mode
GPIO.setup(21,GPIO.OUT)

# Loops the opening and closing of database 
while True:
	# Open database connection
	db = pymysql.connect("localhost","root","password","button")

	# Prepare a cursor object using cursor() method
	cur = db.cursor(pymysql.cursors.DictCursor)
	
	sql = "SELECT power FROM switch WHERE id =2";
	
	# Execute SQL query using execute() method.
	cur.execute(sql)
	
	# Fetch all rows from the database table using cursor's fetchall()
	rows = cur.fetchall()
	for row in rows:
		print(row['power'])
		# Checks if value of power is 1 to turn light on.
		if (row['power'] == 1):
			print("LED on")
			GPIO.output(21,GPIO.HIGH)
		else:
			print("LED off")
			GPIO.output(21,GPIO.LOW)
		time.sleep(1)
			
	# Disconnect from server
	cur.close()
	db.close()

File: sqlLED.php
<html>
<body>

<!-- Form where you can choose 0 to turn light off or 1 to turn light on.  -->

<!-- On submit, send the form-data to a file named "sqlLED.php" --> 

<form action="sqlLED.php" method="post">
<br>

Light: 
<select name="light">
	<option value="0" selected>0</option>
	<option value="1">1</option>
</select>
<br><br>

<input type="submit">
</form>

</body>
</html>

<?php
// Log in information 
$servername = "localhost";
$username = "root";
$password = "password";
$dbname = "button";

// Collect values from form with method = "post"
$light = (int)$_POST["light"];

// Create connection
$conn = new mysqli($servername, $username, $password, $dbname);

// Check connection
if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}

// Insert into database the 0 or 1 values from phpforms.php
$sql = "UPDATE switch SET power = '$light' WHERE id = 2";

// header("location: phpforms.php");

// Checks if new record was created successfully
if ($conn->query($sql) === TRUE) {
    echo "<br>";
    if ($light == 1) {
        echo "Light is on.";
    }
    else {
        echo "Light is off.";
    }
} 

else {
    echo "Error: " . $sql . "<br>" . $conn->error;
}

// Close connection
$conn->close();
?> 
